index.js

```diff
@@ -44,8 +44,28 @@ String.prototype.endswith = function (suffix, start = 0, end = this.length) {
   if (end - start < suffix.length) {
     return false;
   }
-  const substring = this.slice(start, end);
+  substring = this.slice(start, end);
   return substring.endsWith(suffix);
 };
 
+String.prototype.pysplit = function (separator = " ", maxsplit = -1) {
+  if (typeof separator !== 'string') {
+    throw new TypeError("'separator' must be a string");
+  }
+  if (typeof maxsplit !== 'number') {
+    throw new TypeError("'maxsplit' must be a number");
+  }
+  if (separator === "") {
+    throw new Error("ValueError: 'separator' cannot be empty");
+  }
+  if (maxsplit === 0) {
```

Comment:
I'm unsure, but do we need this? (Considering this already points to a `string`)